### PR TITLE
snakemake: added REANA cloud files

### PR DESCRIPTION
**Pull request by mariaFernando**
closes reanahub/reana#530

### CODE REVIEW

**Review by mvidalgarcia — COMMENTED** [review]
Concerning the error in the tests, it looks like we are `grep`ing for an outdated version of `reana-client validate` output [here](https://github.com/reanahub/reana-demo-cms-h4l/blob/b6ed43426c0fda4e5e144e34a85e90e21666ba61/.github/workflows/ci.yml#L29).

We would need something like this:

```console
$ reana-client validate | grep "Valid REANA specification" > /dev/null
```

File: reana.yaml [hunk 1]
```diff
@@ -3,11 +3,12 @@ inputs:
   parameters:
     input: workflow/input.yaml
   directories:
-    - data
     - code
+    - data
+    - workflow/cwl
```
Comment (mvidalgarcia) [inline reana.yaml:8]: There's no `workflow/cwl` path. We would need to arrange the files differently if we want to keep this.

File: workflow/snakemake/input.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+data:
+  class: Directory
+  path: data
+code:
+  class: Directory
+  path: code
```
Comment (mvidalgarcia) [inline workflow/snakemake/input.yaml:6]: What is this file for? Normally we use it to define params, but I don't see any `config[]` reference in the Snakefile.

File: reana.yaml (hunk 1 above)
Comment (mariaFernando) [reply reana.yaml:8]: My bad, forgot to remove this when I changed the directory back

File: workflow/snakemake/input.yaml (hunk 2 above)
Comment (mariaFernando) [reply workflow/snakemake/input.yaml:6]: Added the configuration setting to the Snakefile

**Comment by mvidalgarcia** [comment]
Sharing some data here while trying to run this example in the cluster.

After multiple tests, it looks like [this function call](https://github.com/reanahub/reana-job-controller/blob/39b45e89e7ae06bdb05ba9c8438012fed4b64715/reana_job_controller/job_manager.py#L66) is causing some trouble. In particular [`calculate_file_access_time()` call](https://github.com/reanahub/reana-job-controller/blob/39b45e89e7ae06bdb05ba9c8438012fed4b64715/reana_job_controller/job_manager.py#L133), which takes care of traversing all the files within the workspace, compiling the [last access time](https://docs.python.org/3/library/stat.html#stat.ST_ATIME) (`st_atime`) and returning them, to later create a `JobCache` object and store it in the DB.

Apparently, while traversing the directories, some of the files are removed/renamed/moved (not sure what's the actual reason, maybe they are just temp files or failing symlinks, to be researched), so an exception is raised when trying to get the access time.

After commenting out the `inst.cache_job()` call, the workflow runs as expected. It takes so long to finished but I assume this is normal.

```console
$ reana-client list --filter name=h4l-snake
NAME        RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
h4l-snake   12           2021-09-10T15:40:07   2021-09-10T15:40:16   2021-09-10T16:06:37   finished
h4l-snake   11           2021-09-10T14:40:03   2021-09-10T14:40:19   2021-09-10T14:42:13   failed  
h4l-snake   10           2021-09-10T14:38:02   2021-09-10T14:38:16   2021-09-10T14:39:04   failed  
h4l-snake   9            2021-09-10T14:29:44   2021-09-10T14:29:59   2021-09-10T14:36:24   failed  
h4l-snake   8            2021-09-10T14:28:13   2021-09-10T14:28:28   2021-09-10T14:28:48   failed  
h4l-snake   7            2021-09-10T14:26:56   2021-09-10T14:27:12   2021-09-10T14:27:59   failed  
h4l-snake   6            2021-09-10T14:23:26   2021-09-10T14:23:32   2021-09-10T14:26:34   failed  
h4l-snake   5            2021-09-10T14:14:16   2021-09-10T14:14:23   2021-09-10T14:20:52   failed  
h4l-snake   4            2021-09-10T14:11:24   2021-09-10T14:11:31   2021-09-10T14:12:24   failed  
h4l-snake   3            2021-09-10T14:03:51   2021-09-10T14:03:58   2021-09-10T14:04:41   failed  
h4l-snake   2            2021-09-10T14:01:29   2021-09-10T14:01:36   2021-09-10T14:02:29   failed  
h4l-snake   1            2021-09-10T13:31:16   2021-09-10T13:31:23   2021-09-10T13:36:16   failed  
```

Note the ~26min of execution. But the final output is present.

```console
$ reana-client ls -w h4l-snake --filter name=results                                                                                                                                                                                              
NAME                                           SIZE    LAST-MODIFIED                                                                                                                                                                                                                    
results/mass4l_combine_userlvl3.pdf            18141   2021-09-10T16:06:15                                                                                                                                                                                                              
results/Higgs4L1file.root                      59770   2021-09-10T16:06:05                                                                                                                                                                                                              
results/DoubleMuParked2012C_10000_Higgs.root   54181   2021-09-10T15:54:15                                                                                                                                                                                                              
results/scramdone.txt                          0       2021-09-10T15:40:46
```

And it looks good.

```console
$ reana-client download -w h4l-snake 'results/mass4l_combine_userlvl3.pdf'                                                                                                                                                                        
File results/mass4l_combine_userlvl3.pdf downloaded to /Users/marco/code/reanahub/reana-demo-cms-h4l.
$ open results/mass4l_combine_userlvl3.pdf 
```
![image](https://user-images.githubusercontent.com/8863238/132886928-d00bc74d-0e02-4883-97d9-d15c3f9bfe7e.png)

---

##### TODO
- [ ] Research what's the actual problem when caching files (is the file still present when trying to access it?). Consider adding an exception handling mechanism to avoid these problems (if the file is not there, just skip it and continue). 
- [ ] Analyse jobs **don't** run in parallel. Investigate if we need to add `--cores` or similar. Investigate why [`job_selector`](https://github.com/snakemake/snakemake/blob/26373e66fbaba04fa076aebb24ad869f41f228b9/snakemake/scheduler.py#L492) only return one `analyse` job.
- [ ] Create PR with `r-w-e-snakemake` changes.

**Pull request merged**